src/swift/Sources/Models/TokenFile.swift

```diff
@@ -229,465 +234,430 @@ class TokenFile: Codable {
-            handle(clause: genericWhere)
-        }
+        return processInitializer(defId: defId, attributes: decl.attributes, modifiers: decl.modifiers, kind: decl.kind.textDescription, accessLevel: accessLevel,  genericParam: decl.genericParameterClause, throwsKind: decl.throwsKind, parameterList: decl.parameterList, genericWhere: decl.genericWhereClause)
     }
 
-    private func process(_ decl: FunctionDeclaration) {
-        SharedLogger.debug("Function Declaration")
-        SharedLogger.debug(decl.signature.textDescription)
+    private func process(_ decl: FunctionDeclaration, overridingAccess: AccessLevelModifier? = nil) -> Bool {
+        let accessLevel = decl.modifiers.accessLevel ?? overridingAccess ?? .internal
+        let defId = decl.textDescription
+        let name = decl.name.textDescription
+        return processFunction(name: name, defId: defId, attributes: decl.attributes, modifiers: decl.modifiers, accessLevel: accessLevel, signature: decl.signature, genericParam: decl.genericParameterClause, genericWhere: decl.genericWhereClause)
+    }
 
-        guard publicModifiers.contains(decl.modifiers.accessLevel ?? .internal) else {
-            return
+    private func process(_ decl: SubscriptDeclaration, overridingAccess: AccessLevelModifier? = nil) -> Bool {
+        let accessLevel = decl.modifiers.accessLevel ?? overridingAccess ?? .internal
+        let defId = decl.textDescription
+        return processSubscript(defId: defId, attributes: decl.attributes, modifiers: decl.modifiers, accessLevel: accessLevel, genericParam: decl.genericParameterClause, parameterList: decl.parameterList, resultType: decl.resultType, genericWhere: decl.genericWhereClause)
+    }
+
+    private func process(_ decl: OperatorDeclaration) -> Bool {
+        var kword: String
+        var name: String? = nil
+        var opName: String
+        switch decl.kind {
+        case let .infix(op, ident):
+            kword = "infix"
+            opName = op
+            name = ident?.textDescription
+        case let .prefix(op):
+            kword = "prefix"
+            opName = op
+        case let .postfix(op):
+            kword = "postfix"
+            opName = op
+        }
+        keyword(value: kword)
+        whitespace()
+        keyword(value: "operator")
+        whitespace()
+        text(opName)
+        if let name = name {
+            punctuation(":")
+            whitespace()
+
+            // register type name to make linkable
+            let defId = "operator.\(name)"
+            definitionIds[defId] = defId
+
+            type(name: name, definitionId: defId)
         }
         newLine()
-        if !decl.attributes.isEmpty {
-            handle(attributes: decl.attributes)
-        }
-        handle(modifiers: decl.modifiers)
-        keyword(value: "func")
-        lineIdMarker(definitionId: name)
+        return true
+    }
+
+    private func process(_ decl: PrecedenceGroupDeclaration) -> Bool {
```

Comment:
Why doesn't this one have `overridingAccess`?

Reply:
Same reason as operator. Precedence groups cannot have parents.